src/Microsoft.Extensions.Configuration.Abstractions/IConfiguration.cs

```diff
@@ -36,5 +36,6 @@ public interface IConfiguration
         IEnumerable<IConfigurationSection> GetChildren();
 
         IChangeToken GetReloadToken();
+        IConfigurationRoot Root { get; }
```

Comment:
I guess I still think of these as a node interface.  So I don't think its that bad, since the root of a tree, either has a null parent, or itself.  Am I mistaken to think of IConfiguration in this way?  I still translate it to IConfigurationNode, of which IConfigurationRoot happens to be the root.